oellm/main.py

```diff
@@ -0,0 +1,163 @@
+                    except Exception as e:
+                        logging.debug(
+                            f"Failed to download model {model} from huggingface. Continuing..."
+                        )
+                        logging.debug(e)
+                else:
+                    AutoModelForCausalLM.from_pretrained(model)
+                    model_paths.append(model_path)
+            else:
+                model_paths.append(model)
+
+    tasks = tasks.split(",")
+
+    # cross product of model_paths and tasks into a dataframe
+    df = pd.DataFrame(
+        product(model_paths, tasks, n_shot if isinstance(n_shot, list) else [n_shot]),
+        columns=["model_path", "task_path", "n_shot"],
+    )
+
+    if debug:
+        remaining_queue_capacity = 10
+    else:
+        queue_limit = int(os.environ.get("QUEUE_LIMIT", 250))
+        remaining_queue_capacity = queue_limit - _parse_user_queue_load()
+
+    if remaining_queue_capacity <= 0:
+        logging.warning("No remaining queue capacity. Not scheduling any jobs.")
+        return None
+
+    logging.debug(
+        f"Remaining capacity in the queue: {remaining_queue_capacity}. Number of "
+        f"evals to schedule: {len(df)}."
+    )
+
+    # Save df to temporary CSV file
+    with tempfile.NamedTemporaryFile(
+        mode="w", suffix=".csv", delete=False, delete_on_close=False
+    ) as temp_file:
+        df.to_csv(temp_file.name, index=False)
+        temp_csv_path = temp_file.name
+
+    logging.debug(f"Saved evaluation dataframe to temporary CSV: {temp_csv_path}")
+
+    with open(Path(__file__).parent / "template.sbatch", "r") as f:
+        sbatch_template = f.read()
+
+    # replace the placeholders in the template with the actual values
+    sbatch_script = sbatch_template.format(
+        csv_path=temp_csv_path,
+        max_array_len=max_array_len,
+        array_limit=remaining_queue_capacity - 1,
+        num_jobs=remaining_queue_capacity,
+    )
+
+    logging.debug("--- Generated sbatch script ---")
+    logging.debug(sbatch_script)
+    logging.debug("-------------------------------")
+
+    # Submit the job script to slurm by piping the script content to sbatch
+    try:
+        result = subprocess.run(
+            ["sbatch"],
+            input=sbatch_script,
+            text=True,
+            check=True,
+            capture_output=True,
+        )
+        logging.info("Job submitted successfully.")
+        logging.info(result.stdout)
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Failed to submit job: {e}")
+        logging.error(f"sbatch stderr: {e.stderr}")
+    except FileNotFoundError:
+        logging.error(
+            "sbatch command not found. Please make sure you are on a system with SLURM installed."
+        )
+
+    return df
+
+
+def main():
+    logging.basicConfig(level=logging.DEBUG)
+    """The main entrypoint for the CLI."""
+    CLI({"schedule": schedule_evals}, as_positional=False)
```

Comment:
Maybe we should be specific to eval here as schedule could mean many things?

```suggestion
    CLI({"schedule-eval": schedule_evals}, as_positional=False)
```